src/main/java/seedu/address/logic/parser/IndexIdPair.java

```diff
@@ -131,4 +131,23 @@ public Location getLocationFromPair(Model model) throws CommandException {
             return model.getLocationById(idOpt.get());
         }
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!super.equals(other)) {
+            return false;
+        }
+
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof IndexIdPair)) {
+            return false;
+        }
+
+        IndexIdPair o = (IndexIdPair) other;
+        return prefix.equals(o.prefix);
+    }
```

Comment:
Overriding the equals method from ReadOnlyIndexIdPair will break some tests. This is because some tests make comparisons between IndexIdPair and IndexIdPairStub. Additionally, may I know what would be the significance of an equals method that only compares the prefix? The IndexIdPair also contains two optionals for fields.

Reply:
This equals method first tests equality using the equality method of ReadOnlyIndexIdPair (the first statement). By comparing the prefix in IndexIdPair, we ensure that IdOpt, IndexOpt and the prefix are all equal in this case. Currently, all tests execute successfully, so maybe you could perhaps direct me to the tests you think may be problematic as a result of the change and I'll look into it.

Reply:
Ah I see. Thanks for the explanation. 

The tests that I am referring to are inside VirusTrackerParserTest.java. e.g. parseCommand_deleteLocation(). Within these tests assertEquals is used to compare between the final parsed command which may contain IndexIdPair and the expected command which contains IndexIdPairStub. Considering that IndexIdPairStub doesnt contain a prefix, I was afraid that the tests might fail.

Reply:
I see. Currently, I believe the commands all use ReadOnlyIndexIdPair, so in this case, the tests should still work since the comparisons are between ReadOnlyIndexIdPairs. 

Anyway, for classes that use IndexIdPair, it would be more suitable for them to compare prefixes of the IndexIdPairs as well since having a different prefix can mean that the IndexIdPair are different.

Reply:
I see. Thank you.